File: Backend/app/models/task.py
from tortoise import fields, models
import uuid
from enum import Enum

class TaskStatus(str, Enum):
    TODO = 'todo'
    IN_PROGRESS = 'in_progress'
    REVIEW = 'review'
    DONE = 'done'

class Task(models.Model):
    id = fields.UUIDField(pk=True, default=uuid.uuid4)
    title = fields.CharField(max_length=255)
    description = fields.TextField(null=True)

    status = fields.CharEnumField(enum_type=TaskStatus, default=TaskStatus.TODO)

    assignee = fields.ForeignKeyField(
        'models.User',
        related_name='assigned_tasks',
        on_delete=fields.SET_NULL,
        null=True
    )

    project = fields.ForeignKeyField(
        'models.Project',
        related_name='tasks',
        on_delete=fields.CASCADE
    )

    version = fields.IntField(default=1)

    created_by = fields.ForeignKeyField(
        'models.User',
        related_name='created_tasks',
        on_delete=fields.SET_NULL,
        null=True
    )

    createdAt = fields.DatetimeField(auto_now_add=True)
    updatedAt = fields.DatetimeField(auto_now=True)

    class Meta:
        table = "tasks"

    def __str__(self):
        return f"Task: {self.title} ({self.status})"

File: frontend/src/types/task.ts
export type TaskStatus = "todo" | "in_progress" | "review" | "done"

export interface Task {
  id: string
  title: string
  description?: string | null
  status: TaskStatus
  assignee_ids?: string[]
  assignee_names?: string[]
  project_id: string
  org_id?: string | null
  version: number
  created_by_id?: string | null
  created_by_name?: string | null
  createdAt: string
  updatedAt: string
}

export interface CreateTaskInput {
  title: string
  description?: string | null
  status?: TaskStatus
  assignee_ids?: string[]
}

export interface UpdateTaskInput {
  title?: string | null
  description?: string | null
  status?: TaskStatus | null
  assignee_ids?: string[] | null
  version: number
}

export interface PaginatedTasksResponse {
  items: Task[]
  total: number
  page: number
  page_size: number
  total_pages: number
}
